### DISCUSSION TITLE
Create PRs on https://github.com/microsoft/winget-pkg

### DISCUSSION THREAD

**Discussion by sualeh** [jreleaser/jreleaser#1279] [Q&A]
@aalmiray Does the JReleaser winget packager create PRs on https://github.com/microsoft/winget-pkgs where [SchemaCrawler manifests are hosted](https://github.com/microsoft/winget-pkgs/tree/master/manifests/s/schemacrawler/schemacrawler)?

**Comment by aalmiray** [comment]
It does not create a PR at the moment but it can push a branch (setting `branchPush`) that may be used as a basis for the PR.

**Reply by sualeh** [reply]
But I could not get it to push to https://github.com/microsoft/winget-pkg, under the manifests directory.

**Reply by aalmiray** [reply]
If you have write access to `microsoft/winget-pkgs` then you may create a branch IIRC. If not, then you must push to a fork you control and issue a PR on your own.

**Reply by aalmiray** [reply]
FWIW winget-releaser and komac take care of this (create manifests and push a PR). I have to find how they do it so that I can replicate the behavior.

**Reply by sualeh** [reply]
Thanks, @aalmiray - I will wait until then. There are bots for winget that automatically create updates, so I am not too concerned.

**Comment by jvalkeal** [comment]
I also started to look how this automation into `microsoft/winget-pkgs` might work. I looked `winget-releaser` / `komac` and it has issues with using fine grained access tokens([172](https://github.com/vedantmgoyal2009/winget-releaser/issues/172)). I'm not big fan of komac due to its gh api usage and it assumes that there is at least one existing package PR'd and handled successfully.


Lets first look how jreleaser itself as an example.

1. It's safe to assume that there is an existing fork `jreleaser/winget-pkgs` (fork name could be configurable)
2. JReleaser could automate to create action in `jreleaser/jreleaser-winget` which creates a new branch on `jreleaser/winget-pkgs` with new manifest (PR against community repo can only touch one manifest version)
3. Now the hard part is done and you can create a PR manually or automate even further.

**Reply by aalmiray** [reply]
This is the workflow I currently use for syncing JReleaser to macports. The final step is creating a PR manually because they also expect a given template (with checkboxes and OS specific input) to be filled out.

**Reply by jvalkeal** [reply]
Hah, I never looked those macports repos you have as those indeed look exactly what I wrote.
